intake_bluesky/mongo_embedded.py

```diff
@@ -278,7 +278,8 @@ def search(self, query):
             MongoDB query.
         """
         if query:
-            query = {f"start.{key}": val for key, val in query.items()}
+            query = {f"start.{key}".replace("start.$", "$", 1):
```

Comment:
This is sufficient to pass the tests, but our test coverage is lacking. What about a query like `{'$and', [{'detector': 'pe1}, {'motor': {'$in': ['motor_A', 'motor_B']}}]}`? I think the general solution here will either be to have separate RunStart collection that we treat as a disposable cache or discover a general strategy for transforming queries that are constructed to operate on a collection into queries that operate on a sub-field of documents in that collection.

Reply:
I think I added a decent general solution.